src/index.ts

```diff
@@ -161,4 +161,75 @@ export function lazyload(node: HTMLElement, attributes: Object): ReturnType<Acti
 			observer.unobserve(node);
 		}
 	};
-}
\ No newline at end of file
+}
+
+/**
+ * Prevent current tab from being closed by user
+ * 
+ * Demo: https://svelte.dev/repl/a95db12c1b46433baac2817a0963dc93
+ */
+export const preventTabClose: Action = (_, enabled: boolean) => {
+  const handler = (e: BeforeUnloadEvent) => {
+      e.preventDefault();
+      e.returnValue = '';
+    },
+    setHandler = (shouldWork: boolean) =>
+      shouldWork ?
+        window.addEventListener('beforeunload', handler) :
+        window.removeEventListener('beforeunload', handler);
```

Comment:
TIL about https://developers.google.com/web/updates/2018/07/page-lifecycle-api#the-unload-event looks like you implemented it in the correct way